### PR TITLE
Adding a custom directive 'v-translate' to support HTML translation

### PR DESCRIPTION
**Pull request by kennyki**
After reading and experimenting much with the component, it proves to be too tricky to support HTML content because we cannot get the raw HTML to use as `msgid`.

Thus I came up with a directive that works (at least for me).

Also, I have created a companion library to extract strings from Vue single-file component with template language, i.e. `Pug`. I'm only creating a link reference to the project here.

Feel free to chip in with suggestions.

### CODE REVIEW

**Comment by kemar** [comment]
@kennyki Just a quick word to say I saw your PR, thank you :)

I'll review it as soon as I can (hopefully by the next week).

Just a small comment: could you please remove changes made to the dist files and the version upgrade ([as noted in the contrib guidelines](https://github.com/Polyconseil/vue-gettext/blob/master/README_DEV.md#pull-request-guidelines))?

**Comment by kennyki** [comment]
Thanks @kemar. I have done that.

**Comment by kemar** [comment]
@kennyki looks *almost* good to me. The directive works well. Good job, kudos.

However it doesn't work with the extraction method provided in this plugin (i.e. the Makefile).

Does your companion library allows to extract translations from HTML templates too? Or do you have a method to extract translations from both Vue and HTML files?

I'd like to merge this but I want to add a method to extract translations from the directive before.

File: src/directive.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import interpolate from './interpolate'
+import translate from './translate'
+import Vue from 'vue'
+
+// we have to cache the current language locally
+// to detect if the language has changed and translation should be updated
+let currentLanguage = null
```
Comment (kemar) [inline src/directive.js]: It's not really clear why you have to cache it here, can you elaborate a little more?

**Comment by kennyki** [comment]
@kemar excuse me for being late. Thanks and those are good points about the extraction, I forgot to include them:

> However it doesn't work with the extraction method provided in this plugin (i.e. the Makefile).
> Does your companion library allows to extract translations from HTML templates too? Or do you have a method to extract translations from both Vue and HTML files?

The companion library allows extraction from HTML template, but you will need webpack. It does not extract `v-translate` by default at the moment (I will make it to do that), I was adding it thru the easygettext attributes in my project:

```
node node_modules/vue-webpack-gettext/extract --output static/translate.pot --src src --attrs v-translate
```

For this project's makefile, I will change its line 37 to:

```
gettext-extract --attribute v-translate --quiet --output $@ $(GETTEXT_HTML_SOURCES)
```

File: src/directive.js (hunk 1 above)
Comment (kennyki) [inline src/directive.js]: It's related to the spec: `updates a translation after a language change`

I couldn't find a way to know if the current language has change within the directive's `update` hook, so I needed a cached value to compare against `Vue.config.language` to determine if the translation should be updated. The code is around line 71

**Comment by kemar** [comment]
Ok thank you very much @kennyki 

I've done some small coding style and comment style refactoring :) I added the link to your companion library at the end of the README.

Also I stored the current language in the dataset of the element, I think it works better and it's easier to read: https://github.com/Polyconseil/vue-gettext/blob/8a36abfc336d09b104bc68d1258c56dd7c9ca0fe/src/directive.js

Anyway thank you very much for your great work :)

**Comment by kennyki** [comment]
Sure, no problem. Thanks for your help!

**Pull request merged**